packages/providers/src/pageKind.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { useContext } from 'react';
+import type { SourceFileKind } from 'myst-spec-ext';
+
+const PageKindContext = React.createContext<SourceFileKind | undefined>(undefined);
+
+export function PageKindProvider({
```

Comment:
I would very much like to refactor to `ArticleProvider` from `ReferencesProvider` rather than introduce a new provider. I see why you went this way, but I think that it complicates downstream quite a bit to introduce many smaller providers (already theme/thebe/references/busyscope are a lot to handle).

Happy to take a crack later this week if that doesn't make sense.

Reply:
There's a balance to be struck between having god-providers and having a component tree that looks like you could hang baubles upon it. I sometimes lean towards the latter in my preferences.

Given that we're mainly optimising for ourselves at the moment, I think it's a good nudge to fuse this with `ReferencesProvider`. I've updated the PR!